server/src/main/scala/de/frosner/broccoli/models/ParameterValue.scala

```diff
@@ -0,0 +1,92 @@
+package de.frosner.broccoli.models
+
+import com.typesafe.config.{Config, ConfigValue, ConfigValueType}
+import com.typesafe.config.impl.ConfigInt
+import de.frosner.broccoli.services.{ParameterValueParsingException, TemplateParameterNotFoundException}
+import org.apache.commons.lang3.StringEscapeUtils
+import play.api.libs.json._
+
+import scala.util.{Failure, Success, Try}
+
+object ParameterValue {
+
+  implicit val parameterValueWrites: Writes[ParameterValue] = new Writes[ParameterValue] {
+    override def writes(paramValue: ParameterValue) = paramValue.asJsValue
+  }
+
+  def constructParameterValueFromTypesafeConfig(parameterType: ParameterType,
+                                                configValue: ConfigValue): Try[ParameterValue] =
+    Try {
+      parameterType match {
+        case ParameterType.Raw =>
+          RawParameterValue(configValue.unwrapped().toString)
+        case ParameterType.String =>
+          StringParameterValue(configValue.unwrapped.asInstanceOf[String])
+        case ParameterType.Integer =>
+          IntParameterValue(configValue.unwrapped().asInstanceOf[Int])
+        case ParameterType.Decimal =>
+          DecimalParameterValue(configValue.unwrapped().asInstanceOf[BigDecimal])
+      }
+    }
+
+  def constructParameterValueFromJson(parameterName: String,
+                                      template: Template,
```

Comment:
Why do we need the whole template here? I guess the parameter infos should be enough, or?

The `ParameterValueParsingException` doesn't really need the template ID. It should contain the parameter name and the template abstraction level should be with the caller IMO.